### PR TITLE
Feat/hng 150 add product modal

### LINKED ISSUE #150: [FEAT] Super Admin Add Products Modal Component
### Description:
---

Implement the Add Product Modal components ensuring consistency with ShadCN UI components. This component should adhere strictly to the design specifications provided. Below are the detailed instructions and requirements.

**Requirements:**

1. **Page Components:**
   - Ensure the components match the design specifications provided.

2. **Design Specifications:**
   - Refer to the provided design for dimensions, colors, and layout.

3. **Functionality:**
   - make sure it responds to interaction, it pops up smoothly and closes smoothly 

4. **Accessibility:**
   - Ensure the components is accessible and follow best practices

5. **Testing:**
   - Ensure the component renders correctly with all required elements.

**Deadline: Friday 4pm** 
Reference Components: Inputs, Buttons, and Upload fields

**Acceptance Criteria:**

- [ ] Component Should be pixel perfect as implemented using ShadCN UI.
- [ ] Implement interactions with the component 

**Attachments:**
Screenshot 2024-07-17 at 18 32 01

[Link to the design file](https://www.figma.com/design/VEItfX6St5NSAqqNHImcxD/HNG-Boilerplate-Designs?node-id=2066-12343&m=dev)

- [ShadCN UI Documentation](https://ui.shadcn.com/)

### PR DESCRIPTION
**Pull request by Bwesun**
# Description

**Closes #150**

# Changes proposed

## What were you told to do?
[FEAT] Super Admin Add Products Modal Component https://github.com/hngprojects/hng_boilerplate_nextjs/issues/150

## What did you do?

# Check List (Check all the applicable boxes)

🚨Please review the [contribution guideline](CONTRIBUTING.md) for this repository.

- [X] My code follows the code style of this project.
- [X] This PR does not contain plagiarized content.
- [X] The title and description of the PR is clear and explains the approach.
- [X] I am making a pull request against the **dev branch** (left side).
- [X] My commit messages styles matches our requested structure.
- [X] My code additions will fail neither code linting checks nor unit test.
- [X] I am only making changes to files I was requested to.

# Screenshots/Videos

### CODE REVIEW

File: src/components/ui/textarea.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import * as React from "react"
```
Comment (Justfemi) [inline src/components/ui/textarea.tsx]: Import React from react

File: src/components/ui/label.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
+"use client"
+
+import * as React from "react"
```
Comment (Justfemi) [inline src/components/ui/label.tsx]: Import React from "react" and not "* as React..."

File: src/components/AddProduct/index.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
+import {
```
Comment (Justfemi) [inline src/components/AddProduct/index.tsx:1]: Move the content of this file to the src/components/modals/AddProduct.tsx file

File: src/components/AddProduct/index.tsx [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+  DialogContent,
+  DialogDescription,
+  DialogFooter,
+  DialogHeader,
+  DialogOverlay,
+  DialogTitle,
+  DialogTrigger,
+} from "~/components/ui/dialog";
+import { Button } from "../ui/button";
+import { Input } from "../ui/input";
```
Comment (Justfemi) [inline src/components/AddProduct/index.tsx]: Use ~ for all import (button, input, text area ....) and not relative path

File: package.json
Comment (Justfemi) [inline package.json:1]: Don't edit this file, all you need have been installed. 
If need any package that hasn't been installed, reach out on slack.

**Review by Justfemi — CHANGES_REQUESTED** [review]

**Comment by oreoluwa212** [comment]
`
This branch has conflicts that must be resolved
 to resolve conflicts before continuing.
Conflicting files
package.json
pnpm-lock.yaml
src/app/page.test.tsx
src/app/page.tsx
src/components/modals/PasswordSuccessfulModal/PasswordSuccessfulModal.test.tsx
src/components/modals/PasswordSuccessfulModal/PasswordSuccessfulModal.tsx
src/components/ui/dialog.tsx
src/components/ui/input.tsx
src/components/ui/label.tsx
src/components/ui/textarea.tsx
`

**Comment by Bwesun** [comment]
Please there are files that i have not editted or done anything to. like the first 2 files. the first reviewer said i shouldn't edit the file. furthermore, the rest of the files aren't what i did. i know nothing about them. How do i resolve their conflicts.

**Comment by skeby** [comment]
> Please there are files that i have not editted or done anything to. like the first 2 files. the first reviewer said i shouldn't edit the file. furthermore, the rest of the files aren't what i did. i know nothing about them. How do i resolve their conflicts.

pull from dev, accept all incoming for those files

**Pull request closed without merging**